sub-packages/bionemo-esm2/src/bionemo/esm2/model/finetune/loss.py

```diff
@@ -83,50 +71,37 @@ class ClassifierLossReduction(BERTMLMLossWithReduction):
-        """Calculates the loss within a micro-batch. A micro-batch is a batch of data on a single GPU.
+    ) -> Tuple[torch.Tensor, torch.Tensor, Dict[str, torch.Tensor]]:
+        """Calculates the loss within a micro-batch. A micro-batch is a batch of data on a single GPU. The averaging of the loss is done in super().reduce().
 
         Args:
             batch: A batch of data that gets passed to the original forward inside LitAutoEncoder.
             forward_out: the output of the forward method inside classification head.
-
-        Returns:
-            A tuple where the loss tensor will be used for backpropagation and the dict will be passed to
-            the reduce method, which currently only works for logging.
         """
         targets = batch["labels"].squeeze()  # [b] or [b, s] for sequence-level or token-level classification
+        loss_mask = batch["loss_mask"]
 
         classification_output = forward_out["classification_output"]  # [b, num_class] or [b, s, num_class]
-        # [b, s, num_class] -> [b, num_class, s] to satisfy toke-level input dims for cross_entropy loss
         if classification_output.dim() == 3:
-            classification_output = classification_output.permute(0, 2, 1)
-
-        loss_mask = batch["loss_mask"]  # [b, s]
-
-        cp_size = parallel_state.get_context_parallel_world_size()
-        if cp_size == 1:
-            losses = torch.nn.functional.cross_entropy(classification_output, targets, reduction="none")
-            # token-level losses may contain NaNs at masked locations. We use masked_select to filter out these NaNs
-            if classification_output.dim() == 3:
-                masked_loss = torch.masked_select(losses, loss_mask)
-                loss = masked_loss.sum() / loss_mask.sum()
-            else:
-                loss = losses.mean()  # sequence-level single value classification
+            classification_output = classification_output.permute(1, 0, 2).contiguous()  # change to [s, b, num_class]
+        elif classification_output.dim() == 2:
+            # NOTE: this is for sequence-level classification, we artificially create a sequence dimension to use the same code path as token-level classification
+            classification_output = classification_output.unsqueeze(0)  # change to [1, b, num_class]
+            targets = targets.unsqueeze(1)  # change to [b, 1]
+            loss_mask = torch.ones((targets.shape[0], 1), dtype=loss_mask.dtype, device=loss_mask.device)
         else:
-            raise NotImplementedError("Context Parallel support is not implemented for this loss")
+            raise ValueError(f"Unexpected classification output dimension: {classification_output.dim()}")
 
-        return loss, {"avg": loss}
+        # NOTE: token_logits is [sequence, batch] but labels and other fields, including the loss are [batch, sequence]
+        unreduced_token_loss = unreduced_token_loss_fn(classification_output, targets)  # [b s]
```

Comment:
Good catch that this is already equivalent to F.cross_entropy, just with a different classification "vocabulary".